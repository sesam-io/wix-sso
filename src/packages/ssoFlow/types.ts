import type { Auth0ClientOptions } from "@auth0/auth0-spa-js";

export enum LoginMessageType {
  Login = "auth0:login",
  Logout = "auth0:logout",
  Signup = "auth0:signup",
}

export type RunSSOFlowArgs = {
  auth0ClientOptions: Auth0ClientOptions;
  siteId?: string | "";
};

export type LoadScriptArgs = {
  name: string;
  idAttribute?: string;
  url: string;
};

// The datalayer object is set on the window object by the Google Analytics script
// it can be used to configure GA, set options or send specific events
declare global {
  interface Window {
    dataLayer: unknown[];
  }
}
