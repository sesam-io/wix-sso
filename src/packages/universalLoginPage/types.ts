import { SiteIds } from "./constants";

export type FormType = "login" | "logout";

export type SiteId = keyof typeof SiteIds;

export type Sites = {
  [key: string]: {
    title: string;
    logoUrl?: string;
    logIn: {
      subTitle: string;
    };
    signup: {
      subTitle: string;
    };
  };
};
