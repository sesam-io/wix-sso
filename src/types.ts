export enum LoginMessageType {
  Login = "auth0:login",
  Logout = "auth0:logout",
}

export type LoadScriptArgs = {
  name: string;
  idAttribute?: string;
  url: string;
};
