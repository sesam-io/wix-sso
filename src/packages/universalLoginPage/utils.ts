import { SesamDefaultSite, WixSites } from "./sites";
import { FormType, SiteId } from "./types";

const getWixSiteFn = (wixSites: typeof WixSites) => (siteId: SiteId) => {
  const site = wixSites.find(
    (site) => site.baseId === siteId || site.id === siteId
  );

  if (!site) {
    return SesamDefaultSite;
  }

  return site;
};

export const getWixSite = getWixSiteFn(WixSites);

export const getDefaultPageTitle = (formType: FormType) =>
  formType === "login"
    ? SesamDefaultSite.loginSubTitle
    : SesamDefaultSite.signupSubTitle;
