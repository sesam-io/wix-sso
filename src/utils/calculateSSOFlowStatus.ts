import { SSOFlowStatus } from "../enums";

export type SSOFlowStatusArgs = {
  isAuthenticated: boolean;
  query: string;
};

export const calculateSSOFlowStatus = (args: SSOFlowStatusArgs) => {
  const { isAuthenticated, query } = args;

  if (isAuthenticated) {
    return SSOFlowStatus.LoggedIn;
  }

  if (query.includes("code=") && query.includes("state=")) {
    return SSOFlowStatus.Authenticated;
  }

  return SSOFlowStatus.LoggedOut;
};
