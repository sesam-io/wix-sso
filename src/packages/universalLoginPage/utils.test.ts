// import { BrandedSiteIds, SiteIds } from "./siteIds";
import { SiteId } from "./types";
import { getBaseSiteId, getWixSite } from "./utils";

describe("utils test suite", () => {
  describe("getBaseSiteId test suite", () => {
    it("should return default, when siteId not exist", () => {
      expect(getBaseSiteId()).toMatchSnapshot();
    });

    it("should return poweroffice, when siteId is powerofficego", () => {
      expect(getBaseSiteId("powerofficego")).toMatchSnapshot();
    });

    it("should return poweroffice, when siteId is powerofficego", () => {
      expect(getBaseSiteId("powerofficego")).toMatchSnapshot();
    });

    it("should return poweroffice, when siteId is powerofficego-test", () => {
      expect(getBaseSiteId("powerofficego-test")).toMatchSnapshot();
    });

    it("should return poweroffice, when siteId is camel-cased PowerofficeGo", () => {
      expect(getBaseSiteId("PowerofficeGo")).toMatchSnapshot();
    });

    it("should return tripletex, when siteId is tripletex", () => {
      expect(getBaseSiteId("tripletex")).toMatchSnapshot();
    });

    it("should return tripletex, when siteId is tripletex-test", () => {
      expect(getBaseSiteId("tripletex-test")).toMatchSnapshot();
    });
  });

  describe("getWixSite test suite", () => {
    it("should return default site", () => {
      expect(getWixSite("none-exist-site" as SiteId)).toMatchSnapshot();
    });

    it("should return sesam site", () => {
      expect(getWixSite("sesam")).toMatchSnapshot();
    });

    it("should return Wave site", () => {
      expect(getWixSite("wave")).toMatchSnapshot();
    });

    it("should return SuperOffice site", () => {
      expect(getWixSite("superoffice")).toMatchSnapshot();
    });
  });
});
