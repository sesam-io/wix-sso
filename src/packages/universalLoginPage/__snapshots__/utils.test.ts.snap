// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils test suite getBaseSiteId test suite should return default, when siteId not exist 1`] = `""`;

exports[`utils test suite getBaseSiteId test suite should return poweroffice, when siteId is camel-cased PowerofficeGo 1`] = `"poweroffice"`;

exports[`utils test suite getBaseSiteId test suite should return poweroffice, when siteId is powerofficego 1`] = `"poweroffice"`;

exports[`utils test suite getBaseSiteId test suite should return poweroffice, when siteId is powerofficego 2`] = `"poweroffice"`;

exports[`utils test suite getBaseSiteId test suite should return poweroffice, when siteId is powerofficego-test 1`] = `"poweroffice"`;

exports[`utils test suite getBaseSiteId test suite should return tripletex, when siteId is tripletex 1`] = `"tripletex"`;

exports[`utils test suite getBaseSiteId test suite should return tripletex, when siteId is tripletex-test 1`] = `"tripletex"`;

exports[`utils test suite getWixSite test suite should return SuperOffice site 1`] = `
{
  "baseId": "superoffice",
  "displayPoweredBySesam": true,
  "id": "superoffice",
  "isBrandLogo": true,
  "loginSubTitle": "Log in to SuperOffice DataSync.",
  "logoUrl": "https://cdn.jsdelivr.net/gh/sesam-io/wix-sso@v1.0.127-site-logos/src/packages/siteLogos/superoffice/superoffice-ulp-header-logo.svg",
  "name": "",
  "signupSubTitle": "Sign up to SuperOffice DataSync.",
  "titleClassName": "superofficeLogInTitle",
}
`;

exports[`utils test suite getWixSite test suite should return Wave site 1`] = `
{
  "displayPoweredBySesam": false,
  "id": "wave",
  "isBrandLogo": true,
  "loginSubTitle": "Log in to Making Wave Talk.",
  "logoUrl": "https://cdn.jsdelivr.net/gh/sesam-io/wix-sso@v1.0.127-site-logos/src/packages/siteLogos/wave/making-wave-talk-logo-centered.svg",
  "name": "Wave",
  "signupSubTitle": "Sign up to Making Wave Talk.",
}
`;

exports[`utils test suite getWixSite test suite should return default site 1`] = `
{
  "displayPoweredBySesam": true,
  "id": "sesam",
  "loginSubTitle": "Log in to Sesam Talk.",
  "logoUrl": "https://cdn.jsdelivr.net/gh/sesam-io/wix-sso@v1.0.127-site-logos/src/packages/siteLogos/sesam/sesam-talk-rgb.png",
  "name": "Sesam",
  "signupSubTitle": "Sign up to Sesam Talk to continue to Sesam Talk.",
}
`;

exports[`utils test suite getWixSite test suite should return sesam site 1`] = `
{
  "displayPoweredBySesam": true,
  "id": "sesam",
  "loginSubTitle": "Log in to Sesam Talk.",
  "logoUrl": "https://cdn.jsdelivr.net/gh/sesam-io/wix-sso@v1.0.127-site-logos/src/packages/siteLogos/sesam/sesam-talk-rgb.png",
  "name": "Sesam",
  "signupSubTitle": "Sign up to Sesam Talk to continue to Sesam Talk.",
}
`;
